libs/api/libbpf_program.cpp

```diff
@@ -422,3 +423,77 @@ bpf_prog_bind_map(int prog_fd, int map_fd, const struct bpf_prog_bind_opts* opts
 
     return libbpf_result_err(ebpf_program_bind_map(prog_fd, map_fd));
 }
+
+static int
+__bpf_set_link_xdp_fd_replace(int ifindex, int fd, int old_fd, __u32 flags)
+{
+    if (flags != 0 || old_fd != -1) {
```

Comment:
Did you really want to only support old_fd = -1 ? Also, what happens in `bpf_obj_get_info_by_fd` if old_fd = -1 ?